packages/app/src/backup/services/backup-restore.service.ts

```diff
@@ -0,0 +1,200 @@
+import { RestoreDeviceBackup } from "App/backup/types"
+import { BackupError } from "App/backup/constants"
+
+// DEPRECATED
+import DeviceService from "App/__deprecated__/backend/device-service"
+import DeviceFileSystemAdapter from "App/__deprecated__/backend/adapters/device-file-system/device-file-system-adapter.class"
+
+const timeout = 5000
+const callsMax = 24
+
+export class BackupRestoreService {
+  constructor(
+    private deviceService: DeviceService,
+    private deviceFileSystem: DeviceFileSystemAdapter
+  ) {}
+
+  public async restoreBackup({
+    filePath,
+    backupLocation,
+    token,
+  }: RestoreDeviceBackup): Promise<ResultObject<boolean | undefined>> {
+    const backupId = filePath.split("/").pop() as string
+    const fileData = this.readFile(filePath, token)
+
+    if (!fileData) {
+      return Result.failed(
+        new AppError(
+          BackupError.CannotReadBackupFile,
+          `File: ${filePath} is unreadable`
+        )
+      )
+    }
+
+    const uploadResult = await this.deviceFileSystem.uploadFile({
+      data: this.arrayBufferToBuffer(fileData),
+      targetPath: `${backupLocation}/${backupId}`,
+    })
+
+    if (uploadResult.status !== RequestResponseStatus.Ok) {
+      return Result.failed(
+        new AppError(
+          BackupError.CannotUploadBackupToDevice,
+          uploadResult.error?.message || "Cannot upload backup to device"
+        )
+      )
+    }
+
+    const restoreResult = await this.deviceService.request({
+      endpoint: Endpoint.Restore,
+      method: Method.Post,
+      body: {
+        restore: backupId,
+      },
+    })
+
+    if (restoreResult.status !== RequestResponseStatus.Ok) {
+      return Result.failed(
+        new AppError(
+          BackupError.CannotRestoreBackup,
+          restoreResult.error?.message || "Cannot restore backup"
+        )
+      )
+    }
+
+    const restoreStatus = await this.waitUntilRestoreDeviceFinished(backupId)
+
+    if (restoreStatus.status !== RequestResponseStatus.Ok) {
+      return Result.failed(
+        new AppError(
+          BackupError.RestoreBackupFailed,
+          restoreStatus.error?.message || "Restore failed"
+        )
+      )
+    }
+
+    return Result.success(true)
+  }
+
+  private async waitUntilGetRestoreDeviceStatusNoResponse(
+    id: string,
+    firstRequest = true,
+    index = 0
+  ): Promise<RequestResponse> {
+    if (index === callsMax) {
+      return {
+        status: RequestResponseStatus.Error,
+      }
+    }
+
+    const response = await this.deviceService.request({
+      endpoint: Endpoint.Restore,
+      method: Method.Get,
+      body: {
+        id,
+      },
+    })
+
+    if (response.data?.state === GetRestoreDeviceStatusDataState.Finished) {
+      return {
+        status: RequestResponseStatus.Ok,
+      }
+    }
+
+    if (!firstRequest && response.status === RequestResponseStatus.Error) {
+      return { status: RequestResponseStatus.Ok }
```

Comment:
Just curious - why `Ok` is returned when `status` equals to `Error`?

Reply:
@wojcik92michal I guess, it's needed to prevent errors during the phone wake-up process